mesh/ncmesh.cpp

```diff
@@ -5640,9 +5640,13 @@ bool NCMesh::ZeroRootStates() const
    return true;
 }
 
-void NCMesh::Print(std::ostream &os) const
+void NCMesh::Print(std::ostream &os, std::string const& comments) const
 {
-   os << "MFEM NC mesh v1.0\n\n"
+   os << "MFEM NC mesh v1.0\n\n";
+
+   if (!comments.empty()) { os << '\n' << comments << '\n'; }
+
+   os <<
       "# NCMesh supported geometry types:\n"
```

Comment:
This output formatting seems a bit off. I think this generates a text like this (when `comments` is no-empty):
```ascii
MFEM NC mesh v1.0


<comments>
# NCMesh supported geometry types:
```
and I think it's better to have this:
```ascii
MFEM NC mesh v1.0

<comments>

# NCMesh supported geometry types:
```

Reply:
It already produces the second version, because mesh.cpp:10718 has `\n#\n# MFEM Geometry Types`. It would be easier to verify this if there were a test, but I didn't think a test is appropriate for this simple change. If you make this change in `ex1.cpp`
`mesh.Print(mesh_ofs, "# This is \n# my mesh");`
then you will get the following in `refined.mesh`:
MFEM mesh v1.0

# This is 
# my mesh

#
# MFEM Geometry Types (see mesh/geom.hpp):

Reply:
Fixed in 0e209bbf055e110ade2cd28447ed5210aa21edbd.